app/controllers/templates_controller.rb

```diff
@@ -79,13 +79,19 @@ def auto_complete_controller_name
   end
 
   def preview
-    base = @template.preview_host_collection
-    @host = params[:preview_host_id].present? ? base.find(params[:preview_host_id]) : base.first
+    if params[:id]
+      find_resource
+      base = @template.preview_host_collection
+      @host = params[:preview_host_id].present? ? base.find(params[:preview_host_id]) : base.first
+    else
+      type = type_name_plural
+      @host = params[:preview_host_id].present? ? Host.find(params[:preview_host_id]) : Host.first
```

Comment:
This still needs to use the #preview_host_collection method, as it returns a relation that only permits access to authorised hosts.  If you swap the `@host` and `@template` lines around, I think you can then use `@template.preview_host_collection` still.